File: app/src/main/java/com/brooke/michael/filmbuff/rest/service/RestClient.java
package com.brooke.michael.filmbuff.rest.service;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import retrofit.RestAdapter;
import retrofit.converter.GsonConverter;

/**
 * Created by Michael on 06-Mar-16.
 */
public class RestClient {

    private static final String MOVIE_BASE_URL = "https://api.themoviedb.org/3/";
    public static final String POSTER_BASE_URL = "http://image.tmdb.org/t/p/w300/";

    public static TheMovieDBService getMovieDBClient(){

        Gson gsonConverter = new GsonBuilder()
                .serializeNulls()
                .create();

        RestAdapter adapter = new RestAdapter.Builder()
                .setLogLevel(RestAdapter.LogLevel.FULL)
                .setEndpoint(MOVIE_BASE_URL)
                .setConverter(new GsonConverter(gsonConverter))
                .build();

        return adapter.create(TheMovieDBService.class);
    }

}

File: app/src/main/java/com/brooke/michael/filmbuff/rest/service/RestClient.kt
package com.brooke.michael.filmbuff.rest.service

import com.google.gson.GsonBuilder

import retrofit.RestAdapter
import retrofit.converter.GsonConverter

class RestClient {

    companion object {
        const val MOVIE_BASE_URL = "https://api.themoviedb.org/3/"
        const val POSTER_BASE_URL = "http://image.tmdb.org/t/p/w300/"
        const val API_KEY = "332cc53cb512e33f2caad27caa83d137"

        val instance = RestAdapter.Builder()
                .setLogLevel(RestAdapter.LogLevel.FULL)
                .setEndpoint(MOVIE_BASE_URL)
                .setConverter(GsonConverter(GsonBuilder().serializeNulls().create()))
                .setRequestInterceptor { it.addQueryParam("api_key", "$API_KEY") }
                .build()
                .create(TheMovieDBService::class.java)
    }
}
